core/src/apps/cardano/layout.py

```diff
@@ -164,8 +164,9 @@ async def confirm_transaction(
     pages.append(page1)
 
     page2 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
-    page2.normal("Network:")
-    page2.bold(protocol_magics.to_ui_string(protocol_magic))
+    if is_network_id_verifiable:
+        page2.normal("Network:")
+        page2.bold(protocol_magics.to_ui_string(protocol_magic))
```

Comment:
I am perplexed here: do we only show protocol magic here, and no network id? Or is some of the variable names misleading?

Reply:
the protocol magic is being mapped to a human readable name with the `to_ui_string()` function, and consistency is being ensured when the tx is validated. We don't show the exact number - the user would either see "Mainnet", "Testnet" or "Unknown network"

Reply:
https://github.com/trezor/trezor-firmware/blob/952adc5961e297f7bc402ad1d98456e474f5b744/core/src/apps/cardano/sign_tx.py#L77 - network info validation

https://github.com/trezor/trezor-firmware/blob/4eb5b927c0fc9f53629df328eac663c953c6f707/core/src/apps/cardano/helpers/protocol_magics.py#L14